### ISSUE TITLE
Access Control Running on Login/Logout/Create First User Pages

### ISSUE THREAD

**Issue by MarkAtOmniux** [payloadcms/payload#7330]
### Link to reproduction

https://github.com/MarkAtOmniux/req-user-access-control-issue

### Describe the Bug

Payload v2.24.0 introduces an issue where access control runs on every collection even when not required. For example, the create-first-user page and login, and log out pages run checks on access control.

If your access control checks for req.user, you will see an error on toast appear;
![login-page-error](https://github.com/user-attachments/assets/b414b713-3e4e-4fde-bbe1-3c45dda16c44)
![logout-page-error](https://github.com/user-attachments/assets/c84abc6c-97cb-46fc-aee1-94bc7cae7d2f)

### To Reproduce

Install payload v2.24.0

Add access control checks that make use of the req.user object. For example;

```ts
access: {
		read: ({ req }) => {
			return {
				id: { equals: req.user?.id }, // this now checks req.user.id on the create-first-user and login page
			};
		},
		update: ({ req }) => {
			return {
				id: { equals: req.user?.id }, // this now checks req.user.id on the create-first-user and login page
			};
		},
		delete: ({ req }) => {
			return {
				id: { equals: req.user.id }, // this now checks req.user.id on the create-first-user and login page
			};
		},
		create: ({ req }) => {
			return {
				id: { equals: req.user.id }, // this now checks req.user.id on the create-first-user and login page
			};
		},
		admin: ({ req }) => req.user.roles?.includes('admin'),
	},
```

Navigate to the create-first-user page and an error message will appear telling you that req.user is undefined.

This is especially an issue if you are setting user admin access control settings in a beforeLogin hook, as Payload may show a message telling the user they do not have permission to access the admin panel

### Payload Version

v2.24.1

### Adapters and Plugins

db-mongodb, bundler-webpack

**Comment by sealan** [comment]
To add support to this, I'm running into the same issue on v2.24.1.

I'm not sure if this is related, but the front end now flashes an "Unauthorized" and "You are not allowed to access this page" message between a successful login and loading the Dashboard.

<img width="908" alt="image" src="https://github.com/user-attachments/assets/991579ec-b783-42df-92d6-8d5092fc545e">

**Comment by JarrodMFlesch** [comment]
@MarkAtOmniux @sealan looking into this now.

Thanks @MarkAtOmniux for the detailed repro!

**Comment by JarrodMFlesch** [comment]
@sealan can you share some of your access control functions?

**Comment by sealan** [comment]
Thank you @JarrodMFlesch ! This is now fixed for me in 2.24.2 ❤️

n.b. there is some subtle "flashing" on log out, and I get 3 x `APIError: No User` and `Forbidden: You are not allowed to perform this action.` in the logs.

I did have to update my access controls functions to make them more defensive:

```
import { Access } from 'payload/types';
import { RequestWithTenant } from '../utils/requestWithTenant';
import { arrayOrEmpty } from '../utils/primitive.utils';

const PLATFORM_ADMIN_ROLE = 'platform-admin';
const TENANT_OWNER_ROLE = 'owner';

/**
 * Creates a collection level access function.
 * @param {string} permissionName - The name of the permission to check.
 * @param {Access} [additionalCheckCallback] - An optional additional check callback.
 * @returns {Access} - The access function.
 */
export const createCollectionLevelAccess =
    (permissionName: string, additionalCheckCallback?: Access): Access =>
    async (args) => {
        const req = args.req as RequestWithTenant;

        if (req.user && req.user.roles && req.user.roles === PLATFORM_ADMIN_ROLE) return true;

        // Check if the user has the required permission for the current tenant
        if (!req.user || !req.user.tenants) return false;
        const tenant = arrayOrEmpty(req.user.tenants).find(({ tenant }) =>
            typeof tenant === 'string' ? tenant === req.tenant.id : tenant.id === req.tenant.id,
        );
        if (tenant == null) return false;

        // Check if the user has the required permission for the current tenant
        const hasBasePermission =
            tenant.role === TENANT_OWNER_ROLE || tenant.permissions?.[permissionName];

        // If an additional callback is provided, check its return value as well
        const hasAdditionalPermission = additionalCheckCallback
            ? await additionalCheckCallback({ req })
            : true;

        // Return true or hasAdditionalPermission(maybe AccessResult) only if both the base permission check and the additional check pass
        const result = hasBasePermission && hasAdditionalPermission;
        return result;
    };
```

_ignore the gross util functions pls_

**Comment by JarrodMFlesch** [comment]
@sealan there is still flashing? Can you take a screen recording?
